connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,112 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.*;
+import java.lang.reflect.InvocationTargetException;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConnectionTest {
+    private static Connection connection;
+    private static ByteArrayOutputStream innerOutputStream;
+    private static int originalNumberOfBytes;
+    private static final SocketAddress address = mock(SocketAddress.class);
+    private static final Message messageToReceive = new Message(TEXT, "MessageToReceive");
+    private static final Message messageToSend = new Message(TEXT, "MessageToSend");
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void setUpConnectionObjectWithByteStreams() throws IOException {
+        ByteArrayOutputStream outerOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream outerOOS = new ObjectOutputStream(outerOutputStream)) {
+            outerOOS.writeObject(messageToReceive);
+        }
+        ByteArrayInputStream innerInputStream = new ByteArrayInputStream(outerOutputStream.toByteArray());
+
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        try (ObjectOutputStream oos = new ObjectOutputStream(buffer)) {
+            oos.writeObject(new Message(TEXT, "InOrderNotToGetEOFException"));
+        }
+        byte[] rawData = buffer.toByteArray();
+        originalNumberOfBytes = rawData.length;
+
+        innerOutputStream = new ByteArrayOutputStream();
+        innerOutputStream.write(rawData);
+
+        Socket socket = mock(Socket.class);
+        when(socket.getOutputStream()).thenReturn(innerOutputStream);
+        when(socket.getInputStream()).thenReturn(innerInputStream);
+        when(socket.getRemoteSocketAddress()).thenReturn(address);
+
+        connection = new Connection(socket);
+    }
+
+    public void setUpConnectionWithFileStream() throws IOException {
+        FileOutputStream buffer = new FileOutputStream("src\\test\\" +
+                "java\\com\\netcracker\\TestResource");
+
+        try (ObjectOutputStream oos = new ObjectOutputStream(buffer)) {
+            oos.writeObject(messageToSend);
+        }
+
+        FileInputStream innerInputStream = new FileInputStream("src\\test\\" +
+                "java\\com\\netcracker\\TestResource");
```

Comment:
There is no other way to test close() method, because ByteArrayOutput(Input)Stream cannot be closed. If I call close() method of BAO(I)Stream, it didn't work because it shouldn't do anything according to the documentation. So I have to use other streams, like FileOutput(Input)Stream.